### DISCUSSION TITLE
Hosting on Raspberry Pi

### DISCUSSION THREAD

**Discussion by werfgyfudi7** [jagrosh/MusicBot#807] [Questions]
How would I go about this? I've seen the 24hr hosting page but I'm not really sure how to do it. Also, what kind of specs for the Pi are necessary?

**Comment by MichailiK** [comment]
Any Raspberry Pi will likely work, since JMusicBot is not resource-intensive, at least when only few servers use it at the same time.

You will need to follow the [Setup Guide](https://github.com/jagrosh/MusicBot/wiki/Setup), the jists of it are that you will need to [install Java](https://www.google.com/search?q=Install%20Java%20Raspberry%20Pi), [download](https://github.com/jagrosh/MusicBot/releases/download/0.3.5/JMusicBot-0.3.5.jar) the MusicBot, by using [any means to download files](https://www.google.com/search?q=Download+files+Linux) to a Raspberry Pi, and start the bot by using:

java -Dnogui=true -jar JMusicBot-**\*insert version here\***.jar

**Reply by TauchMe** [reply]
It is strange that although it is working in my raspberry 32-bit, it shows me that I'm not using a supported Java version.

![image](https://user-images.githubusercontent.com/83581543/133507328-04a4461a-35f2-47af-bed9-0eece8397957.png)

**Reply by MichailiK** [reply]
32-bit Java isn't supported as there are issues. For example, changing the volume to anything other than 100 will break MusicBot & won't play any music on that server.

**Comment by YJSoft** [comment]
Although it's still in beta, You can install 64bit(arm64) version of java with arm64 raspberry pi os.
Or you can always use aarch64 Ubuntu.

https://downloads.raspberrypi.org/raspios_arm64/images/raspios_arm64-2021-05-28/
